### PR TITLE
MGMT-7555 - Fix `make destroy_nodes` isn't working 

### PR DESCRIPTION
**Pull request by eliorerz**
- Create package form virsh_cleanup and remove duplicated file. 
- Create target for destroying and creating onprem env

I'm not sure what to do with `virsh_cleanup.py` and `delete_nodes.py`. They share the same logic
/cc @osherdp @tsorya

### CODE REVIEW

File: src/virsh_cleanup/delete_nodes.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+
+@utils.on_exception(message="Failed to delete kube api resources", silent=True)
+def delete_kube_api_resources_from_namespaces(namespace):
+    kube_api_client = service_client.ClientFactory.create_kube_api_client()
+
+    if namespace != "all":
+        return delete_kube_api_resources_for_namespace(
+            kube_api_client=kube_api_client,
+            name=f"{args.cluster_name or consts.CLUSTER_PREFIX}-{namespace}",
```
Comment (osherdp) [inline src/virsh_cleanup/delete_nodes.py:143]:
```suggestion
            name=args.cluster_name or f"{consts.CLUSTER_PREFIX}-{namespace}",
```

File: src/virsh_cleanup/delete_nodes.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            namespace=namespace,
+        )
+
+    v1 = CoreV1Api(kube_api_client)
+    for namespace in v1.list_namespace():
+        return delete_kube_api_resources_for_namespace(
+            kube_api_client=kube_api_client,
+            name=f"{args.cluster_name or consts.CLUSTER_PREFIX}-{namespace}",
```
Comment (osherdp) [inline src/virsh_cleanup/delete_nodes.py:151]:
```suggestion
            name=args.cluster_name or f"{consts.CLUSTER_PREFIX}-{namespace}",
```

File: src/virsh_cleanup/delete_nodes.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            namespace=namespace,
+        )
+
+
+@utils.on_exception(message="Failed to delete nodes", silent=True, errors=(FileNotFoundError,))
+def main():
+    if args.kube_api:
+        delete_kube_api_resources_from_namespaces(args.namespace)
+
+    if args.delete_all:
```
Comment (osherdp) [inline src/virsh_cleanup/delete_nodes.py:161]: maybe we should rename to ``delete_all_resources``?

File: src/virsh_cleanup/delete_nodes.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        _delete_virsh_resources()
+        return
+
+    if args.namespace == "all":
+        delete_clusters_from_all_namespaces()
+        return
```
Comment (osherdp) [inline src/virsh_cleanup/delete_nodes.py:167]: hmm, this is after we check ``delete_all``?
I'm confused

File: src/virsh_cleanup/delete_nodes.py [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        type=strtobool,
+        nargs="?",
+        const=True,
+        default=False,
+    )
+
+    oc_utils.extend_parser_with_oc_arguments(parser)
+    args = parser.parse_args()
+    if not args.kube_api:
+        args.namespace = ""
```
Comment (osherdp) [inline src/virsh_cleanup/delete_nodes.py:203]:
```suggestion
        args.namespace = ""

```

File: src/virsh_cleanup/__main__.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import argparse
+
+from .virsh_cleanup import DEFAULT_SKIP_LIST, clean_virsh_resources, log
+
+
+def _get_parsed_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Description of your program")
```
Comment (osherdp) [inline src/virsh_cleanup/__main__.py:7]: ?

File: src/virsh_cleanup/__main__.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-a", "--all", help="Clean all virsh resources", action="store_true")
+    group.add_argument("-m", "--minikube", help="Clean minikube resources", action="store_true")
```
Comment (osherdp) [inline src/virsh_cleanup/__main__.py:10]:
```suggestion
    group.add_argument("-m", "--minikube", help="Clean only minikube resources", action="store_true")
```

File: src/virsh_cleanup/__main__.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    group.add_argument("-sm", "--skip-minikube", help="Clean all but skip minikube resources", action="store_true")
```
Comment (osherdp) [inline src/virsh_cleanup/__main__.py:11]:
```suggestion
    group.add_argument("--skip-minikube", help="Clean all but skip minikube resources", action="store_true")
```

File: Makefile [hunk 9]
```diff
@@ -255,6 +255,9 @@ validate_namespace:
 
 run: validate_namespace deploy_assisted_service deploy_ui
 
+run_onprem:
+	DEPLOY_TARGET=onprem make validate_namespace deploy_assisted_service deploy_ui
```
Comment (osherdp) [inline Makefile:259]: ``validate_namespace deploy_assisted_service deploy_ui`` is the same as ``run``, no?
In general, I think we'd better have less number of makefile targets, as it's gets harder to maintain a larger number of them
so ``make run DEPLOY_TARGET=onprem`` should be simple enough imho

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/virsh_cleanup/delete_nodes.py [hunk 10]
```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+
+import argparse
+import os
+import re
+import shutil
+from distutils.util import strtobool
```
[the remark this answers is not in the document]
Comment (eliorerz) [reply src/virsh_cleanup/delete_nodes.py:8]: Only part of distutils will become deprecated on python 3.10 (e.g. `distutils.sysconfig`).
I think util module is not going to be deprecated (and I couldn't find any alternative for it).

[the remark this answers is not in the document]
Comment (osherdp) [reply src/virsh_cleanup/delete_nodes.py:8]: okay, we can handle it later

File: Makefile (hunk 9 above)
Comment (eliorerz) [reply Makefile:259]: I remember changing it to `DEPLOY_TARGET=onprem make run`
But I wondered about this too.

**Review by osherdp — APPROVED** [review]

**Comment by osherdp** [comment]
/override ci/prow/e2e-metal-assisted-capi
shouldn't even run, considering the changes

**Pull request merged**